### ISSUE TITLE
Need better error reporting when a port is already in use.

### ISSUE THREAD

**Issue by jordansissel** [logstash-plugins/logstash-input-beats#268]
When Beats input cannot start correctly because the port is already in use, Logstash logs a complex error message including a stack trace.

Ideally, the error Logstash presents should be more actionable and less noisy for the user.

Additionally, because this is is an uncaught exception handled by the input, logstash will keep trying to restart this plugin. This may be an OK behavior, but it continuously reports a long stack trace for the same problem over and over each plugin restart.

----

How to reproduce:

```
% bin/logstash -e 'input { beats { port => 5000 } beats { port => 5000 } }'
```

Alternately, use the port in another process and start with just one beats input:

```
% nc -l 5000
% bin/logstash -e 'input { beats { port => 5000 } }'
```

The log output looks like this:

```

[2017-11-12T07:09:55,189][INFO ][org.logstash.beats.Server] Starting server on port: 5000
[2017-11-12T07:10:02,106][ERROR][logstash.pipeline        ] A plugin had an unrecoverable error. Will restart this plugin.
  Plugin: <LogStash::Inputs::Beats port=>5000, id=>"2b63ba237fc0eea34f6adffb76f5787c15908767-2", enable_metric=>true, codec=><LogStash::Codecs::Plain id=>"plain_ab1080fb-31c6-45ed-9738-d3e26b0061a4", enable_metric=>true, charset=>"UTF-8">, host=>"0.0.0.0", ssl=>false, ssl_verify_mode=>"none", include_codec_tag=>true, ssl_handshake_timeout=>10000, congestion_threshold=>5, target_field_for_codec=>"message", tls_min_version=>1, tls_max_version=>1.2, cipher_suites=>["TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384", "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384", "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256", "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256", "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384", "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384", "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256", "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256"], client_inactivity_timeout=>60>
  Error: Address already in use

[2017-11-12T07:10:03,112][INFO ][org.logstash.beats.Server] Starting server on port: 5000
Nov 12, 2017 7:10:03 AM io.netty.channel.AbstractChannel$AbstractUnsafe register
WARNING: Force-closing a channel whose registration task was not accepted by an event loop: [id: 0xc8275696]
java.util.concurrent.RejectedExecutionException: event executor terminated
java.util.concurrent.RejectedExecutionException: event executor terminated
        at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:840)
        at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:342)
        at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:335)
        at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:765)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.register(AbstractChannel.java:475)
        at io.netty.channel.SingleThreadEventLoop.register(SingleThreadEventLoop.java:80)
        at io.netty.channel.SingleThreadEventLoop.register(SingleThreadEventLoop.java:74)
        at io.netty.channel.MultithreadEventLoopGroup.register(MultithreadEventLoopGroup.java:85)
        at io.netty.bootstrap.AbstractBootstrap.initAndRegister(AbstractBootstrap.java:330)
        at io.netty.bootstrap.AbstractBootstrap.doBind(AbstractBootstrap.java:281)
        at io.netty.bootstrap.AbstractBootstrap.bind(AbstractBootstrap.java:277)
        at io.netty.bootstrap.AbstractBootstrap.bind(AbstractBootstrap.java:259)
        at org.logstash.beats.Server.listen(Server.java:69)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.jruby.javasupport.JavaMethod.invokeDirectWithExceptionHandling(JavaMethod.java:436)
        at org.jruby.javasupport.JavaMethod.invokeDirect(JavaMethod.java:300)
        at org.jruby.java.invokers.InstanceMethodInvoker.call(InstanceMethodInvoker.java:35)
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:134)
        at org.jruby.ast.CallNoArgNode.interpret(CallNoArgNode.java:60)
        at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
        at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
        at org.jruby.evaluator.ASTInterpreter.INTERPRET_METHOD(ASTInterpreter.java:74)
        at org.jruby.internal.runtime.methods.InterpretedMethod.call(InterpretedMethod.java:182)
        at org.jruby.internal.runtime.methods.DefaultMethod.call(DefaultMethod.java:203)
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:168)
        at org.jruby.ast.CallOneArgNode.interpret(CallOneArgNode.java:57)
        at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
        at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
        at org.jruby.ast.RescueNode.executeBody(RescueNode.java:221)
        at org.jruby.ast.RescueNode.interpret(RescueNode.java:116)
        at org.jruby.ast.EnsureNode.interpret(EnsureNode.java:96)
        at org.jruby.ast.BeginNode.interpret(BeginNode.java:83)
        at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
        at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
        at org.jruby.evaluator.ASTInterpreter.INTERPRET_METHOD(ASTInterpreter.java:74)
        at org.jruby.internal.runtime.methods.InterpretedMethod.call(InterpretedMethod.java:182)
        at org.jruby.internal.runtime.methods.DefaultMethod.call(DefaultMethod.java:203)
        at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:168)
        at org.jruby.ast.FCallOneArgNode.interpret(FCallOneArgNode.java:36)
        at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
        at org.jruby.evaluator.ASTInterpreter.INTERPRET_BLOCK(ASTInterpreter.java:112)
        at org.jruby.runtime.Interpreted19Block.evalBlockBody(Interpreted19Block.java:206)
        at org.jruby.runtime.Interpreted19Block.yield(Interpreted19Block.java:194)
        at org.jruby.runtime.Interpreted19Block.call(Interpreted19Block.java:125)
        at org.jruby.runtime.Block.call(Block.java:101)
        at org.jruby.RubyProc.call(RubyProc.java:300)
        at org.jruby.RubyProc.call(RubyProc.java:230)
        at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:103)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by AbdulHaseebHussain** [comment]
This issue has been raised many times previously.

I think this could be averted by catching a BindException within the [Server.listen](https://github.com/logstash-plugins/logstash-input-beats/blob/b5a714916eaa9ed6ab7c2a2efab167a298b6af19/src/main/java/org/logstash/beats/Server.java#L53) method with a log warning message indicating that the port is already in use.

Unfortunately I could not test these changes as I could not install the plugin due to dependency errors.

**Comment by praseodym** [comment]
This issue can be closed now that #289 is merged.

**Comment by robbavey** [comment]
Thanks @praseodym
